slider/replay.py

```diff
@@ -152,6 +154,64 @@ def _consume_actions(buffer):
     return out
 
 
+def _ms(t):
+    return t.total_seconds() * 1e3
+
+
+def _within(p1, p2, d):
+    return (p1.x - p2.x) ** 2 + (p1.y - p2.y) ** 2 < d ** 2
+
+
+def _pressed(datum):
+    return datum.key1 or datum.key2 or datum.mouse1 or datum.mouse2
+
+
+def _process_circle(obj, rdatum, hw, scores):
+    out_by = abs(_ms(rdatum.offset - obj.time))
+    if out_by < hw.hit_300:
+        scores["perfects"].append(obj)
+    elif out_by < hw.hit_100:
+        scores["goods"].append(obj)
+    elif out_by < hw.hit_50:
+        scores["bads"].append(obj)
+
+
+def _process_slider(obj, rdata, head_hit, rad, scores):
+    ticks = obj.tick_points
+    missed_points = 0
+    if not head_hit:
+        missed_points += 1
+        scores["sliderbreaks"].append(obj)
```

Comment:
hmm i'm not sure about an object, i was treating this more as something that sorts the hit objects in the map into categories and returning new objects instead feels a bit weird. Doesn't it also go into 300 or 100 or 50 or miss later? Although since it now must have non-zero `missed_points` it can't go into 300s anymore.